lib/util/s3.ts

```diff
@@ -5,7 +5,10 @@ import { parseUrl } from './url';
 let s3Instance: S3 | undefined;
 export function getS3Client(): S3 {
   if (!s3Instance) {
-    s3Instance = new S3({});
+    s3Instance = new S3({
+      ...(process.env.AWS_ENDPOINT && { endpoint: process.env.AWS_ENDPOINT }),
+      ...(process.env.AWS_PATH_STYLE && { forcePathStyle: true }),
+    });
```

Comment:
do we need additional info in the docs section?
does current `s3://bucket-name` handling sufficient for minIO etc. ?

Reply:
> do we need additional info in the docs section?

yes, we should.

> does current `s3://bucket-name` handling sufficient for minIO etc. ?

yes, maybe `AWS_PATH_STYLE ` isn't required. i can test again when you've added the changes. i already reverted those to do other stuff 😅

Reply:
🤔 This will have side effects when using maven s3 deps

https://github.com/renovatebot/renovate/blob/229643f7e426192ba1bcb53d5ad6cf11bf304341/lib/modules/datasource/maven/util.ts#L116

Reply:
do we absolutely need to use the same s3 instance?

Reply:
no, maybe we should allow some additional / alternate credentials for cache case?
I don't have any maven project with s3 dependencies, but we've also other datasources which are using aws api.

Maybe we should name those extra env `S3_ENDPOINT` and `S3_PATH_STYLE` instead ? or `S3_CACHE_...` ?

Reply:
OK, i need `forcePathStyle` for minio:
```
 WARN: RepoCacheS3.read() - failure (repository=renovate-reproductions/17125-renovate-badges-test)
       "err": {
         "errno": -3008,
         "code": "ENOTFOUND",
         "syscall": "getaddrinfo",
         "hostname": "renovate.s3.domain.test",
         "$metadata": {"attempts": 1, "totalRetryDelay": 0},
         "message": "getaddrinfo ENOTFOUND renovate.s3.domain.test",
         "stack": "Error: getaddrinfo ENOTFOUND renovate.s3.domain.test\n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26)"
       }
```

Reply:
If for example we use both maven s3 deps and another S3 compatible services such as `minIO` at once, would that mean we have to access two different endpoints? if this is the case, shouldn't we use separate s3 instances ?

just to confirm, does `minIO` uses the same s3 credentials and only supplies its out endpoint?
because as I understand it, every `s3Clinet` instance will use the same credentials as runtime env is the same.

Reply:
i think we should leave as is now, as we curretnly can't support multiple endpoints, as the `AWS_` env would be applied to all anyways. 

So i think we can assume if a self-hosted user uses maven s3, he would also use the same credentials for the renovate cache on same aws reagion or a self-hosted s3 (if that is somehow possible).

We can think about different AWS credentials later if that is somehow required for somebody.

I think we should only rename the S3 specific variables to make them more clear. 

```suggestion
    s3Instance = new S3({
      ...(process.env.S3_ENDPOINT && { endpoint: process.env.S3_ENDPOINT }),
      ...(process.env.S3_FORCE_PATH_STYLE && { forcePathStyle: true }),
    });
```

maybe also add the `RENOVATE_X_`  prefix and add docs to:
https://github.com/gabriel-ladzaretti/renovate/blob/e98b90d54a7a09dc9006b189fc7f7936a53eead1/docs/usage/self-hosted-experimental.md